### PR TITLE
[17.01] add release notes

### PR DESCRIPTION
**Pull request by martenson**
Please help with curating these @galaxyproject/core - this is our work being presented. 

**Every committer has push access to this branch on my fork.**

### CODE REVIEW

File: doc/source/releases/17.01.rst [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+Enhancements
+-------------------------------
+
+.. major_feature
+
+
+.. feature
+
+* New User Preferences User Interface
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: `@anuprulez <https://github.com/anuprulez>`__ and

File: doc/source/releases/17.01.rst [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3118`_
+* Per user total walltime limit.
+  (thanks to `@interactomix <https://github.com/interactomix>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/interactomix/maxf130/g

File: doc/source/releases/17.01.rst [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3217`_
+* Add API to install resolver tool dependencies
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3222`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: This was backported to 16.10 before its release and already announced in the corresponding release notes, I'd remove it or move it to the `enhancement` section.

File: doc/source/releases/17.01.rst [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+* Replace CasperJS browser tests with Selenium tests.
+  `Pull Request 3239`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Not sure this is of enough general interest to be a feature.

File: doc/source/releases/17.01.rst [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+
+* added STL datatype
+  (thanks to `@ThomasWollmann <https://github.com/ThomasWollmann>`__).
+  `Pull Request 2679`_
+* Add cached dependency manager
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3106`_
+* Support for gzipped fastq formats
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3145`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: This should be moved to `feature` IMO.

File: doc/source/releases/17.01.rst [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+* Version 17.01.dev
+  `Pull Request 3192`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Remove, it's minor (I tagged it as such now).

File: doc/source/releases/17.01.rst [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+* update rename dataset action to accept . in repeat input parameters
+  (thanks to `@scholtalbers <https://github.com/scholtalbers>`__).
+  `Pull Request 3197`_
+* Fix user table in admin view
+  (thanks to `@erasche <https://github.com/erasche>`__).
+  `Pull Request 3200`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: I'd move this to `bug` section.

File: doc/source/releases/17.01.rst [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3218`_
+* Add EDAM topic(s) and operation(s) elements to XSD file.
+  (thanks to `@hmenager <https://github.com/hmenager>`__).
+  `Pull Request 3221`_
+* Fix configuration API admin checking and tests.
+  `Pull Request 3228`_
+* Further tighten up GG format.
+  `Pull Request 3233`_
+* change conda channel order
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/galaxyproject/bgruening/g

File: doc/source/releases/17.01.rst [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3286`_
+* Fix SLURM job re-submission.
+  `Pull Request 3290`_
+* Ready Integration Testing for Prime Time
+  `Pull Request 3291`_
+* lexists vs. exists
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3301`_
+* Bring in galaxyproject/galaxy-lib`#36
+  <https://github.com/galaxyproject/galaxy/issues/36>`__ from @bgruening.
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s|galaxy/issues/36|galaxy-lib/pull/36|

File: doc/source/releases/17.01.rst [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+  (thanks to `@erasche <https://github.com/erasche>`__).
+  `Pull Request 3340`_
+* Remove workflow testing stuff in test/functional and references.
+  `Pull Request 3342`_
+* Cache depencencies on the fly when first used
+  (thanks to `@abretaud <https://github.com/abretaud>`__).
+  `Pull Request 3348`_
+* Fuzzy search for tools search
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3356`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/bgruening/anuprulez/g

Also, move to feature?

File: doc/source/releases/17.01.rst [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+* Limit adjusting Galaxy's Python environment to legacy tools.
+  `Pull Request 3364`_
+* Remove defunct script run_selenium_tests.sh.
+  `Pull Request 3367`_
+* Workflow run display improvements - esp. for inputs.
+  `Pull Request 3369`_
+* Expand selenium test coverage for workflows.
+  `Pull Request 3371`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: What about collecting all CasperJS and Selenium test entries in a single one with all the PR links?

File: doc/source/releases/17.01.rst [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3379`_
+* Added image datatypes
+  (thanks to `@ThomasWollmann <https://github.com/ThomasWollmann>`__).
+  `Pull Request 3380`_
+* Test cases & small enhancement to workflow renaming PJAs.
+  `Pull Request 3381`_
+* Fix typo
+  (thanks to `@pvanheus <https://github.com/pvanheus>`__).
+  `Pull Request 3384`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move to `bug` section.

File: doc/source/releases/17.01.rst [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+* Resolve Conda Dependencies All at Once
+  `Pull Request 3391`_
+* Markdown datatype support.
+  `Pull Request 3399`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Remove, this was reverted in https://github.com/galaxyproject/galaxy/pull/3408

File: doc/source/releases/17.01.rst [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+* Backport `#3395 <https://github.com/galaxyproject/galaxy/issues/3395>`__
+  `Pull Request 3397`_
+* Escapes some variables
+  (thanks to `@yhoogstrate <https://github.com/yhoogstrate>`__).
+  `Pull Request 3403`_
+* Remoteuser Logout Fix.
+  `Pull Request 3405`_
+* Revert "Markdown datatype support."
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
+  `Pull Request 3408`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Remove together with the note about #3399.

File: doc/source/releases/17.01.rst [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3384`_
+* Resolve Conda Dependencies All at Once
+  `Pull Request 3391`_
+* Markdown datatype support.
+  `Pull Request 3399`_
+* extend API version information
+  (thanks to `@manabuishii <https://github.com/manabuishii>`__).
+  `Pull Request 3411`_
+* Target Miniconda3 when installing Conda.
+  `Pull Request 3419`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move to `feature`?

File: doc/source/releases/17.01.rst [hunk 16]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3324`_
+* More IPython to Jupyter changes
+  (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
+  `Pull Request 3333`_
+* Update UCSC_SERVERS list to include new(ish?) soe subdomain
+  `Pull Request 3345`_
+* Fix explicit Docker container resolution.
+  `Pull Request 3346`_
+* Masthead Webhook bug fix
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/bgruening/anuprulez/g

File: doc/source/releases/17.01.rst [hunk 17]
```diff
@@ -0,0 +1,417 @@
...
+  `Pull Request 3367`_
+* Workflow run display improvements - esp. for inputs.
+  `Pull Request 3369`_
+* Expand selenium test coverage for workflows.
+  `Pull Request 3371`_
+* Use a newer conda version by default
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
+  `Pull Request 3378`_
+* Add diamond database as datatype
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: and `@bebatut <https://github.com/bebatut>`__

**Review by nsoranzo — COMMENTED** [review]
Thanks @martenson! I've added some corrections and suggestions.

**Comment by martenson** [comment]
@nsoranzo Can you please directly modify and commit your changes? You can do so on GitHub directly:
https://github.com/martenson/galaxy/blob/17.01_release_notes/doc/source/releases/17.01.rst

I believe nobody will dispute your corrections or suggestions on release notes.

edit: I will happily do so myself if it is inconvenient for you or you don't have time, just tell me please.

**Comment by nsoranzo** [comment]
@martenson Thanks for doing it and sorry, I didn't have time.

**Review by nsoranzo — APPROVED** [review]

File: doc/source/releases/17.01_announce.rst [hunk 18]
```diff
@@ -8,14 +8,21 @@ January 2017 Galaxy Release (v 17.01)
...
+  within the Galaxy's workspace. This will enable your Galaxy to install tool dependencies using Conda
+  e.g. scientific `BioConda <https://bioconda.github.io/>`__ channel.
+
+**New interface for user preferences**
+  User preferences menu has been reworked for clarity and consistency.
+  Thanks to `@anuprulez <https://github.com/anuprulez>`__ and `@bgruening <https://github.com/bgruening>`__.
+  Implemented in `Pull Request 3118`_.
+
+**Support for gzipped fastq formats**
+  You can now upload compressed files without being unpacked. Existing tools will work as before and future
```
Comment (martenson) [inline doc/source/releases/17.01_announce.rst]: need to specify which formats.

File: doc/source/releases/17.01_announce.rst [hunk 19]
```diff
@@ -8,14 +8,21 @@ January 2017 Galaxy Release (v 17.01)
...
+  After updating to 17.01 and upon starting Galaxy the Conda package manager will be installed automatically
+  within the Galaxy's workspace. This will enable your Galaxy to install tool dependencies using Conda
+  e.g. scientific `BioConda <https://bioconda.github.io/>`__ channel.
+
+**New interface for user preferences**
+  User preferences menu has been reworked for clarity and consistency.
+  Thanks to `@anuprulez <https://github.com/anuprulez>`__ and `@bgruening <https://github.com/bgruening>`__.
+  Implemented in `Pull Request 3118`_.
+
+**Support for gzipped fastq formats**
```
Comment (nsoranzo) [inline doc/source/releases/17.01_announce.rst]: s/gzipped fastq/compressed FASTQ/

File: doc/source/releases/17.01_announce.rst (hunk 18 above)
Comment (nsoranzo) [inline doc/source/releases/17.01_announce.rst]: s/compressed files/compressed (with gzip or bzip2) FASTQ files/

**Review by nsoranzo — COMMENTED** [review]
Thanks @martenson, brilliant!

**Comment by martenson** [comment]
Thanks for your help @nsoranzo !

This is now ready for review @galaxyproject/core

File: doc/source/releases/17.01_announce.rst [hunk 20]
```diff
@@ -10,17 +10,20 @@ Highlights
...
 **New interface for user preferences**
   User preferences menu has been reworked for clarity and consistency.
   Thanks to `@anuprulez <https://github.com/anuprulez>`__ and `@bgruening <https://github.com/bgruening>`__.
   Implemented in `Pull Request 3118`_.
 
-**Support for gzipped fastq formats**
-  You can now upload compressed files without being unpacked. Existing tools will work as before and future
-  tools will be able to consume archives and save space in your quota.
+**Support for compressed FASTQ formats**
+  You can now upload compressed (with gzip or bzip2) FASTQ files without them being unpacked automatically (as it was until now).
```
Comment (nsoranzo) [inline doc/source/releases/17.01_announce.rst]: Should we also specify that this applies only if you explicitly assign the corresponding datatype during file upload (sniffing disabled by default)?

Comment (martenson) [reply doc/source/releases/17.01_announce.rst]: done

File: doc/source/releases/17.01_announce.rst [hunk 21]
```diff
@@ -3,8 +3,50 @@
...
+  Please see the details at the admin `documentation <https://docs.galaxyproject.org/en/master/admin/conda_faq.html>`__
+
+**New interface for user preferences**
+  User preferences menu has been reworked for clarity and consistency.
+  Thanks to `@anuprulez <https://github.com/anuprulez>`__ and `@bgruening <https://github.com/bgruening>`__.
+  Implemented in `Pull Request 3118`_.
+
+**Support for compressed FASTQ formats**
+  You can now upload compressed (with gzip or bzip2) FASTQ files without them being unpacked automatically (as it was until now).
+  For this to work you have to select the proper type when uploading (e.g. instead of ``fastqsanger`` you select ``fastqsanger.gz`` for compressed files).
```
Comment (mvdbeek) [inline doc/source/releases/17.01_announce.rst]: s/``fastqsanger.gz``/``fastqsanger.gz or fastqsanger.bz2``/

File: doc/source/releases/17.01_announce.rst [hunk 22]
```diff
@@ -3,8 +3,50 @@
...
+  Existing tools will work as before and future tools will be able to consume archives and save space in your quota.
+  Thanks to `@mvdbeek <https://github.com/mvdbeek>`__.
```
Comment (mvdbeek) [inline doc/source/releases/17.01_announce.rst]: `@pvanheus @abretaud @ashok` also contributed commits in that PR,
and I think we should also mention https://github.com/galaxyproject/galaxy/pull/3514 and https://github.com/galaxyproject/galaxy/pull/3510

File: doc/source/releases/17.01_announce.rst (hunk 21 above)
Comment (martenson) [reply doc/source/releases/17.01_announce.rst]: This is an example, I do not think we need to list more possibilities.

**Comment by mvdbeek** [comment]
Thanks @martenson, I didn't know we exclude galaxy-lab members.

**Comment by martenson** [comment]
@mvdbeek Yeah, if we didn't it would have been too much text I believe. We tend to do some things worthy of release notes too. 😋 

Also we want to highlight everybody else.

**Comment by jmchilton** [comment]
@martenson As promised a did a major pass at the release notes. Feel free to back out of anything you disagree with and we can discuss it more. I may make another pass as well but this first one was fairly significant.

File: doc/source/releases/17.01.rst [hunk 23]
```diff
@@ -9,163 +9,124 @@
...
-.. feature
-
-* Target Miniconda3 when installing Conda.
+* Add cached dependency manager (was back-ported to 16.10 as well)
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3106`_
+* Fix Conda channel order
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3237`_
+* Target Miniconda3 when initialized new Conda installs.
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: `when initialized` doesn't sound right to me.

**Comment by martenson** [comment]
splendid, thanks @jmchilton

File: doc/source/releases/17.01.rst [hunk 24]
```diff
@@ -9,163 +9,124 @@
...
+  `Pull Request 3285`_, `Pull Request 3286`_, `Pull Request 3291`_,
+  `Pull Request 3342`_, `Pull Request 3367`_, `Pull Request 3381`_,
+  `Pull Request 3239`_, `Pull Request 3371`_, `Pull Request 3190`_,
+  `Pull Request 3264`_, `Pull Request 3288`_, `Pull Request 3387`_,
+  `Pull Request 3400`_, `Pull Request 3401`_, `Pull Request 3228`_,
+  `Pull Request 3365`_, `Pull Request 3368`_, `Pull Request 3370`_,
+  `Pull Request 3482`_, `Pull Request 3185`_
+* Fix to use lexists instead exists in dependency resolver code
   (thanks to `@bgruening <https://github.com/bgruening>`__).
   `Pull Request 3301`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move this to the "dependency resolver" section above?

File: doc/source/releases/17.01.rst [hunk 25]
```diff
@@ -9,163 +9,124 @@
...
-* Bring in galaxyproject/galaxy-lib`#36
+* Bring in fix from galaxy-lib related to Biocontainers and mulled
   <https://github.com/galaxyproject/galaxy-lib/pull/36>`__ from @bgruening.
   `Pull Request 3304`_
-* Update datatype converter tools
+* Large cleanup of datatype converter tools
   (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
   `Pull Request 3305`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move this to the "datatypes" section above?

File: doc/source/releases/17.01.rst [hunk 26]
```diff
@@ -9,163 +9,124 @@
...
-* Adds a 'make charts' makefile target to make it more obvious how to build
+* Adds a ``make charts`` Makefile target to make it more obvious how to build
   these.
   `Pull Request 3306`_
-* Make it easier to use job re-submission conditions.
-  `Pull Request 3319`_
-* Addition of IGV display for genome fasta file
+* Implement IGV display for FASTA datatypes
   (thanks to `@Delphine-L <https://github.com/Delphine-L>`__).
   `Pull Request 3327`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move this to the "datatypes" section above?

File: doc/source/releases/17.01.rst [hunk 27]
```diff
@@ -9,163 +9,124 @@
...
-* IE enhancement
+* Add an option to specify the port when connecting clients to IEs.
   (thanks to `@bgruening <https://github.com/bgruening>`__).
   `Pull Request 3328`_
 * Allow admins to see API keys under REMOTE_USER
   (thanks to `@erasche <https://github.com/erasche>`__).
   `Pull Request 3338`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Maybe we can also group the API PRs.

File: doc/source/releases/17.01.rst [hunk 28]
```diff
@@ -9,163 +9,124 @@
...
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3237`_
+* Target Miniconda3 when initialized new Conda installs.
   `Pull Request 3419`_
-* Per user total walltime limit.
+* Cache dependencies on the fly when first used
+  (thanks to `@abretaud <https://github.com/abretaud>`__).
+  `Pull Request 3348`_
+* Use a newer conda version by default
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/galaxyproject/bgruening/g

File: doc/source/releases/17.01.rst [hunk 29]
```diff
@@ -174,19 +135,11 @@ Fixes
...
   `Pull Request 3490`_
-* Attempt to fix broken integration tests on Jenkins.
-  `Pull Request 3365`_
-* Fix auto creation of XUnit reports for Jenkins.
-  `Pull Request 3368`_
-* Fix automatic handling of Jenkins BUILD_NUMBER in Dockerized tests.
-  `Pull Request 3370`_
 * Fix dependency display and api
   (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
   `Pull Request 3427`_
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: Move this to the "dependency resolver" section above?

**Review by nsoranzo — COMMENTED** [review]
Thanks @jmchilton, great work!

File: doc/source/releases/17.01.rst (hunk 29 above)
Comment (jmchilton) [reply doc/source/releases/17.01.rst]: Sorry - I was working on this as you were commenting. I merged this in with the generic fixes just now I think.

File: doc/source/releases/17.01.rst (hunk 28 above)
Comment (jmchilton) [reply doc/source/releases/17.01.rst]: Caught this myself on the second pass.

File: doc/source/releases/17.01.rst (hunk 24 above)
Comment (jmchilton) [reply doc/source/releases/17.01.rst]: I actually moved this into the bug fixes section. I think that it is fine down there. We could list it under "misc" fixes for dependency resolvers also - do you feel strongly about this?

Comment (nsoranzo) [reply doc/source/releases/17.01.rst]: That's fine, thanks!

File: doc/source/releases/17.01.rst [hunk 30]
```diff
@@ -0,0 +1,331 @@
...
+  `Pull Request 3378`_
+* Resolve Conda dependencies all at once to allow Conda to select compatible
+  builds.
+  `Pull Request 3391`_
+* Implement dependency resolver mapping files.
+  `Pull Request 3444`_
+* Add API to install resolver tool dependencies
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3222`_
+* Various other dependency resolver fixes and improvements.
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst:40]: Marius should be thanked as author of 2 of the following PRs.

File: doc/source/releases/17.01.rst [hunk 31]
```diff
@@ -0,0 +1,331 @@
...
+  `Pull Request 3403`_, `Pull Request 3416`_, `Pull Request 3448`_
+* Fix import order and Python3 compatibility for lib/galaxy/tools/
+  (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
+  `Pull Request 3431`_
+* Fix SLURM job re-submission.
+  `Pull Request 3290`_
+* Dependency resolver documentation fixes
+  (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
+  `Pull Request 3490`_
+* Fix directory extistence checks in dependency resolver code
```
Comment (nsoranzo) [inline doc/source/releases/17.01.rst]: s/extistence/existence/

**Comment by jmchilton** [comment]
Thanks for the [cat herding](https://www.fallon.com/case/cat-herders-2/) @martenson!

**Pull request merged**